src/xrpld/app/tx/detail/VaultWithdraw.cpp

```diff
@@ -307,11 +292,16 @@ VaultWithdraw::doApply()
         // else quietly ignore, account balance is not zero
     }
 
-    auto const dstAcct = [&]() -> AccountID {
-        if (ctx_.tx.isFieldPresent(sfDestination))
-            return ctx_.tx.getAccountID(sfDestination);
-        return account_;
-    }();
+    auto const dstAcct = ctx_.tx[~sfDestination].value_or(account_);
+    if (!vaultAsset.native() &&               //
+        dstAcct != vaultAsset.getIssuer() &&  //
+        dstAcct == account_)
+    {
+        if (auto const ter = addEmptyHolding(
+                view(), account_, mPriorBalance, vaultAsset, j_);
+            ter && ter != tecDUPLICATE)
```

Comment:
nit:  I find it a bit more clear by explicitly specifying that we are looking for non-tesSuccess code
```suggestion
            !isTesSuccess(ter) && ter != tecDUPLICATE)
```